R/sampleVarSpec.R

```diff
@@ -284,7 +284,11 @@ sampleVarSpecServer <- function(id, # nolint
   assert_string(label_modal_title)
 
   moduleServer(id, function(input, output, session) {
-    start_col_data <- eventReactive(experiment_name(), {
+
+    to_observe <- reactive({list(experiment_name(), original_data())})
+
+    start_col_data <- eventReactive(to_observe(), {
```

Comment:
Doesn't have to trigger on `experiment_name`. `experiment_name` isn't used in this reactive and also not needed to keep module working
```suggestion
    start_col_data <- reactive({
```

Comutation of `final_data` depends on several reactives and `current_combination` which depends on `experment_name`. Red link denotes that not needed

![merge_data-Page-4](https://github.com/insightsengineering/teal.modules.hermes/assets/6959016/c48b567d-9f93-4a05-9ff8-515221acaee6)

Reply:
You are right, I am now a bit confused about the original design.

Reply:
I am not so sure. The col data variables generally depend on which experiment was selected. Maybe not in the current data example in the app but in general they can be different